src/map/chrif.c

```diff
@@ -324,6 +324,9 @@ bool chrif_save(struct map_session_data *sd, int flag) {
 	if( sd->save_quest )
 		intif->quest_save(sd);
 
+	if (VECTOR_LENGTH(sd->storage))
```

Comment:
Saving an empty storage is a case that should be handled normally. A storage can currently have 0 items in it, and we still want to save it.
This happens when a character has items in its storage, logs in, then removes all the items from its storage. In that case we still want to save an empty storage, or the items will be duplicated.